Library/Homebrew/shims/mac/super/curl-config

```diff
@@ -0,0 +1,35 @@
+#!/bin/bash -p
+
+# HOMEBREW_LIBRARY is set by bin/brew
+# HOMEBREW_CURL_CONFIG is set by brew.sh
+# shellcheck disable=SC2154
+source "${HOMEBREW_LIBRARY}/Homebrew/shims/utils.sh"
+
+# Check if HOMEBREW_CURL_CONFIG is set, fallback to system curl-config if not.
+curl_config_exec="${HOMEBREW_CURL_CONFIG:-/usr/bin/curl-config}"
+
+# Intercept --features and append HTTP2 if needed
+if [[ "$1" == "--features" ]]
+then
+  output="$("${curl_config_exec}" "$@")"
+
+  # Check if HTTP2 is missing but supported
+  if [[ ! "${output}" =~ HTTP2 ]]
+  then
+    curl_version="$("${HOMEBREW_CURL:-curl}" -V)"
+    if [[ "${curl_version}" =~ HTTP/2 ]]
+    then
+      output="${output} HTTP2"
```

Comment:
Can you explain why we'd want to claim HTTP2 support is present when `curl` says its not?

Reply:
> `curl-config --features` did not report HTTP2 prior to macOS 14

even though curl does support http2

Reply:
Gotcha, thanks. I'd be tempted to just use Homebrew's `curl` in that case rather than have a shim. Can we use `uses_from_macos "curl", since:` instead to handle macOS 13 and below?

Reply:
yeah, that should also work.

Reply:
@MikeMcQuaid that hack (`uses_from_macos "curl", since: :sonoma`) would actually enable the linkage against system curl, see [this build log](https://github.com/Homebrew/homebrew-core/actions/runs/11872759852/job/33086872512).

In other words, I still feel like it is better to patch the curl-config report. I know this would be only for ventura builds. Let me know what you think.

cc @Bo98 

(we have already merged the linked PRs on the homebrew-core side)

Reply:
> @MikeMcQuaid that hack (`uses_from_macos "curl", since: :sonoma`) would actually enable the linkage against system curl, see [this build log](https://github.com/Homebrew/homebrew-core/actions/runs/11872759852/job/33086872512).

@chenrui333 on which macOS versions would it link against/not link against system `curl`? Thanks!

Reply:
> I know this would be only for ventura builds.

For one, non-latest macOS version this seems overkill, personally, but I'm open to thoughts from other maintainers.

Reply:
yeah, this only happens for ventura builds, for sonoma/sequoia builds, it works fine.

Reply:
understand. 👍 😄